### PR TITLE
Add estimated gas log in publish command

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related #20

### CODE REVIEW

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -347,9 +347,18 @@ program
       }\nTo alter these settings use the parameters '--max-fee-per-gas', '--max-priority-fee-per-gas', '--gas-limit'.`
     );
 
+    const signer = p.signers[0];
+    let contract;
+    let recipient;
```
Comment (mohammadranjbarz) [inline packages/cli/src/index.ts:352]: @noahlitvin  I didnt know how should I fill `contract`, `recipient` and is `contract.estimateGas.publish` right or not?

**Comment by mohammadranjbarz** [comment]
@noahlitvin  It's ready for review now

Screen Shot 1402-04-29 at 15 29 10
Screen Shot 1402-04-29 at 15 29 50

https://goerli.etherscan.io/tx/0x9a16c2f3db1f23aae8f9e6a96fc4aea503d0e49f2d5e9cd70a78983ca1fba704

**Review by dbeal-eth — CHANGES_REQUESTED** [review]
Is there any other data we can print about the Multicall besides hust the gas estimate? For example, can we also print out the number of calls in the Multicall, the selected gas price, perhaps the list of packages included? I know it's not included on the original issue spec but this is not the only thing about publish loggigng that needs a revisit

**Comment by mohammadranjbarz** [comment]
> Is there any other data we can print about the Multicall besides hust the gas estimate? For example, can we also print out the number of calls in the Multicall, the selected gas price, perhaps the list of packages included? I know it's not included on the original issue spec but this is not the only thing about publish loggigng that needs a revisit

@dbeal-eth  I don't have much experience in solidy so I don't know how much data we can have, you and @noahlitvin  @aminlatifi  can help me, if there is other data please tell me then I will try to achieve them

**Comment by mohammadranjbarz** [comment]
@dbeal-eth 
I added more details in the logs, how is it now?

Screen Shot 1402-05-02 at 19 25 43

File: packages/builder/src/registry.ts [hunk 2]
```diff
@@ -318,4 +317,22 @@ export class OnChainRegistry extends CannonRegistry {
 
     return url === '' ? null : url;
   }
+
+  private async logMultiCallEstimatedGas(datas: any, overrides: any): Promise<void> {
+    try {
+      const estimatedGas = await this.contract.estimateGas.multicall(datas, overrides);
+      console.log(`\nEstimated gas: ${estimatedGas}`);
+      const gasPrice = await this.provider?.getGasPrice();
```
Comment (aminlatifi) [inline packages/builder/src/registry.ts:325]: Override values must be observed.

**Review by aminlatifi — CHANGES_REQUESTED** [review]

**Review by dbeal-eth — APPROVED** [review]
😆 ks great, see Amin comment

**Comment by mohammadranjbarz** [comment]
@dbeal-eth  @aminlatifi 

Now I consider input maxFeePerGas

* When the user doesn't enter `--max-fee-per-gas`
Screen Shot 1402-05-03 at 12 43 26

* When the user enters `--max-fee-per-gas`
Screen Shot 1402-05-03 at 12 43 42

File: packages/builder/src/registry.ts [hunk 3]
```diff
@@ -318,4 +317,21 @@ export class OnChainRegistry extends CannonRegistry {
 
     return url === '' ? null : url;
   }
+
+  private async logMultiCallEstimatedGas(datas: any, overrides: Overrides): Promise<void> {
+    try {
+      const estimatedGas = await this.contract.estimateGas.multicall(datas, overrides);
+      console.log(`\nEstimated gas: ${estimatedGas}`);
+      const gasPrice = (overrides.maxFeePerGas as BigNumber) || (await this.provider?.getGasPrice());
```
Comment (aminlatifi) [inline packages/builder/src/registry.ts:325]: `overrides` can have `gasPrice` property as well which takes precedence over the network default value (obtained by provider.getGasPrice)

**Review by aminlatifi — CHANGES_REQUESTED** [review]

**Review by dbeal-eth — APPROVED** [review]

**Pull request merged**